maas-schemas/schemas/core/components/common.json

```diff
@@ -117,6 +117,11 @@
       "description": "Whim only deep link to localhost and freely defined view - not a complete URI validation",
       "type": "string",
       "pattern": "^(whim):\\/\\/\\/[^\\s/$.?#].[^\\s]*$"
+    },
+    "emptyObject": {
+      "type": "object",
+      "properties": {},
+      "additionalProperties": false
```

Comment:
I don't see why this is needed? 😅 Can't it be defined where it's needed?

Reply:
Because it's used in two other components? Therefore it makes sense to factor it out into a common definition.

It's used in configurator.json and customerSelection.json.
The reason emnptyObject is needed in the anyOf, is because confirguratorV1 is quite loosely defined, meaning technically an empty object matches. It was causing false-positive errors in validation, because it seems AJV was struggling to decide between configuratorV1 and configuratorV2. Adding this emptyObject as one of the possibilities means that the validator gives the expected results (I added tests for these cases).

Reply:
ok. What I wanted to ask is about the actual empty object, not about the reusability of the components.
It's good to know about this AJV issue